src/dawn/AST/ASTStmt.h

```diff
@@ -100,24 +139,29 @@ class Stmt : public std::enable_shared_from_this<Stmt> {
   void setID(int id) { statementID_ = id; }
 
 protected:
+  // copy assignment
   void assign(const Stmt& other) {
+    DAWN_ASSERT_MSG((checkSameDataType(other)), "Trying to assign Stmt with different data type");
     kind_ = other.kind_;
     loc_ = other.loc_;
+    *data_ = *other.data_;
```

Comment:
maybe actually the clone was necessary. are we slicing here?

Reply:
Yes, need clone. Fix with next commit.